components/renku_data_services/session/shipwright_core.py

```diff
@@ -0,0 +1,111 @@
+"""Business logic for ShipWright resources."""
+
+import logging
+from typing import TYPE_CHECKING
+
+from renku_data_services.session import constants, crs, models
+from renku_data_services.session.k8s_client import ShipwrightClient
+
+if TYPE_CHECKING:
+    from renku_data_services.app_config.config import BuildsConfig
+
+
+async def update_build_status(
+    build: models.Build, shipwright_client: ShipwrightClient | None
+) -> models.ShipWrightBuildStatusUpdate | None:
+    """Update the status of a build by pulling the corresponding BuildRun from ShipWright.
+
+    Note: this method will update `build` in place.
+    """
+    if shipwright_client is None:
+        logging.warning("ShipWright client not defined, BuildRun refresh skipped.")
+        return None
```

Comment:
issue: just do not accept None for the `ShipwrightClient` parameter passed in here. Then you can remove this whole section. There is no other reason to pass in the Shipwright client as None here. The same goes for every other function in this file.

Reply:
Done in [705fef2](https://github.com/SwissDataScienceCenter/renku-data-services/pull/625/commits/705fef2a1e8db539f9678d57f309ceb8ecc9eff9).